File: Python3/ransom_note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(magazine) < len(ransomNote): return False
        
        hashmap = {}
        
        for w in ransomNote:
            if w not in magazine: return False
            if w in magazine and w not in hashmap:
                hashmap[w] = magazine.count(w) - 1
                continue
            if w in magazine and w in hashmap and hashmap[w] != 0:
                hashmap[w] -= 1
                continue
            if w in magazine and w in hashmap and hashmap[w] == 0:
                return False
            
        return True

File: C/ransom_note.c
bool canConstruct(char * ransomNote, char * magazine){
    if(strlen(ransomNote) > strlen(magazine)) return false;
    
    int char_count[26] = {0};
    
    for(int i = 0; i < strlen(magazine); i++) {
        char_count[magazine[i] - 'a'] ++;
    }

    for(int i = 0; i < strlen(ransomNote); i++) {
        if(char_count[ransomNote[i]-'a'] <= 0) return false;
        else char_count[ransomNote[i]-'a'] --;
    }

    return true;
}
